bitcoin/util/vector.h

```diff
@@ -56,4 +56,22 @@ inline V Cat(V v1, const V& v2)
     return v1;
 }
 
+/** Concatenate any number of vectors into the first one, moving elements. */
+template<typename V, typename... Vs>
+inline V CatMany(V v, const Vs&&... vs)
```

Comment:
With the `const` before `Vs`, I don't think moving will have any effect.

Reply:
A more generic version that decides per argument whether moving is possible:

```c++
template<typename V, typename... Vs>
inline V CatMany(V v, Vs&&... vs)
{
    v.reserve(v.size() + (vs.size() + ...));
    ([&](Vs&& arg) {
        if constexpr (std::is_lvalue_reference_v<Vs>) {
            v.insert(v.end(), arg.begin(), arg.end());
        } else {
            v.insert(v.end(), std::make_move_iterator(arg.begin()), std::make_move_iterator(arg.end()));
        }
    }(std::forward<Ts>(vs)), ...);
    return v;
}
```


}

Reply:
Hmm, of these approaches lack the nice automatic type inference `Cat` had.

Reply:
Yes, that's why i made it a `CatMany` instead of extending `Cat`. I'll go with your `BuildScript` suggestion instead for now.